simulation/halsim_print/src/halsim_print/cpp/main.cpp

```diff
@@ -0,0 +1,31 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2016-2017 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include <iostream>
+
+#include <HAL/Ports.h>
+
+#include "HALSimPrint.h"
+#include "PrintPWM.h"
+
+/* Currently, robots never terminate, so we keep a single static object
+   and it is never properly released or cleaned up. */
+static HALSimPrint halsim;
+
+extern "C" {
+#if defined(WIN32) || defined(_WIN32)
+__declspec(dllexport)
+#endif
+    int HALSIM_InitExtension(void) {
+  std::cout << "Print Simulator Initializing." << std::endl;
```

Comment:
Should use llvm::raw_ostream.

Reply:
Please refer to prior comment; I think the benefit is not worth the added complexity.